src/pymor/vectorarrays/block.py

```diff
@@ -178,8 +189,9 @@ def __eq__(self, other):
     def dim(self):
         return sum(subspace.dim for subspace in self.subspaces)
 
-    def zeros(self, count=1, reserve=0):
-        return BlockVectorArray([subspace.zeros(count=count, reserve=reserve) for subspace in self.subspaces], self)
+    def zeros(self, count=1, reserve=0, dtype=None):
+        return BlockVectorArray([subspace.zeros(count=count, reserve=reserve, dtype=None)
+                                 for subspace in self.subspaces], self)
```

Comment:
Ok, to ask the other way around: Are there situations where it would be crucial to have a `dtype` argument for `zeros` or `empty`? As far as I can see, there are mainly two possibilities:
1. automatic type promotion is not allowed for some reason
2. automatic type promotion would be to expensive (probably in terms of memory)